src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -116,16 +116,19 @@ public static SocketError Connect(SafeCloseSocket handle, byte[] peerAddress, in
                 IntPtr.Zero);
             return errorCode == SocketError.SocketError ? GetLastSocketError() : SocketError.Success;
         }
-
         public static SocketError Send(SafeCloseSocket handle, IList<ArraySegment<byte>> buffers, SocketFlags socketFlags, out int bytesTransferred)
         {
+            const int StackThreshold = 80; // arbitrary limit to avoid too much space on stack
```

Comment:
We might want to lower this, maybe to 40.  From a quick glance at WSABuffer and GCHandle, it looks like if we actually hit this threshold, we'd be at ~2K of stack space, which is more than we generally like to consume with stackallocs.  I've also generally not seen more than a few buffers passed, so shrinking this shouldn't be particularly problematic.  Maybe 16.  Then you could also change the below from `stackalloc WSABuffer[count]` to `stackalloc WSABuffer[StackThreshold]`... using a const value is more efficient than a dynamic one, as it allows the JIT to generate code specific to that value.

Reply:
makes sense; how about `Select`? that is just `IntPtr` - maybe 64?